roles/ovirt-host-deploy-firewalld/tasks/main.yml

```diff
@@ -0,0 +1,71 @@
+---
+- name: Fetch info about VDSM
+  command: bash -c "rpm -qi vdsm | grep -oE 'Version\\s+:\\s+[0-9\\.]+' | awk '{print $3}'"
+  changed_when: false
+  register: vdsm_version
+
+- name: Check if VDSM version is supported for FirewallD
+  fail:
+    msg: "VDSM version {{ vdsm_version }} is not supported for FirewallD."
+  when:
+    - "host_deploy_firewall_type == 'FIREWALLD'"
+    - "vdsm_version | version_compare('4.20', '>=')"
+
+- block:
+    - name: Stop iptables service if running
+      service:
+        name: iptables
+        state: stopped
+        enabled: no
+      ignore_errors: yes
+
+    - name: Remove iptables-service if exists
+      package:
+        name: iptables-services
+        state: absent
+
+    - name: Check if firewalld is installed
+      package:
+        name: firewalld
+        state: present
+    
+    - name: Check if firewalld is runnning
+      service:
+        name: firewalld
+        state: started
+        enabled: yes
+    
+    - name: Enable VDSM port
+      firewalld:
+        port: "{{ host_deploy_vdsm_port }}/tcp"
```

Comment:
We store in database `vdsm_port`, I am unsure if it was ever configurable and it's not now, so this can be removed.

```
engine=# select port from vds_static ;
 port  
-------
 54321
(1 row)
```